Recruit-Training/Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -60,12 +60,29 @@ bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) con
     QStringList sections = line.split(BATDATA_DELIMITER);
 
     QString timeString = sections.at(0);
+
+    if (sections.size() != COLUMNS)
+    {
+        return false;
+    }
+
+    bool validVoltage = false;
+    bool validCurrent = false;
+
     batteryData.time = QTime::fromString(timeString, STRING_TIME_FORMAT);
+    batteryData.voltage = sections.at(1).toDouble(&validVoltage);
+    batteryData.current = sections.at(2).toDouble(&validCurrent);
 
-    batteryData.voltage = sections.at(1).toDouble();
+    bool validTime = QTime::isValid(batteryData.time.hour(), batteryData.time.minute(), batteryData.time.second() + batteryData.time.msec()/1000);
```

Comment:
If lines are too long, just separate the line into multiple using the ENTER key. Fix all instances of long lines.